main.tf

```diff
@@ -0,0 +1,293 @@
+
+      content {
+        rule_variables {
+          key = "HOME_NET"
+          ip_set {
+            definition = var.home_net
+          }
+        }
+      }
+    }
+
+    stateful_engine_options {
+      rule_order = "STRICT_ORDER"
+    }
+
+    stateful_default_actions = ["aws:drop_established"]
+
+    dynamic "stateful_rule_group_reference" {
+      for_each = { for k, v in var.managed_rule_groups : k => v if v.enabled && v.drop_mode == true }
+
+      content {
+        priority     = stateful_rule_group_reference.value.priority
+        resource_arn = "arn:aws:network-firewall:${data.aws_region.default.name}:aws-managed:stateful-rulegroup/${stateful_rule_group_reference.key}StrictOrder"
+      }
+    }
+
+
+    dynamic "stateful_rule_group_reference" {
+      for_each = { for k, v in var.managed_rule_groups : k => v if v.enabled && v.drop_mode == false }
+
+      content {
+        priority     = stateful_rule_group_reference.value.priority
+        resource_arn = "arn:aws:network-firewall:${data.aws_region.default.name}:aws-managed:stateful-rulegroup/${stateful_rule_group_reference.key}StrictOrder"
+
+        override {
+          action = "DROP_TO_ALERT"
+        }
+      }
+    }
+
+    dynamic "stateful_rule_group_reference" {
+      for_each = length(local.sorted_fqdn_rules) > 0 ? { create : true } : {}
+
+      content {
+        priority     = var.fqdn_rule_group_priority
+        resource_arn = aws_networkfirewall_rule_group.fqdn_rules[0].arn
+      }
+    }
+
+    dynamic "stateful_rule_group_reference" {
+      for_each = length(local.sorted_ip_rules) > 0 ? { create : true } : {}
+
+      content {
+        priority     = var.ip_rule_group_priority
+        resource_arn = aws_networkfirewall_rule_group.ip_rules[0].arn
+      }
+    }
+  }
+}
+
+resource "aws_networkfirewall_rule_group" "fqdn_rules" {
+  count = length(local.sorted_fqdn_rules) > 0 ? 1 : 0
+
+  capacity = var.fqdn_rules_capacity
+  name     = "${var.name}-fqdn-rules"
+  type     = "STATEFUL"
+
+  rule_group {
+    stateful_rule_options {
+      rule_order = "STRICT_ORDER"
+    }
+
+    rule_variables {
+      dynamic "ip_sets" {
+        for_each = local.sorted_fqdn_rules
+
+        content {
+          key = "src_ip_${ip_sets.key}"
+
+          ip_set {
+            definition = ip_sets.value.source_ip
+          }
+        }
+      }
+
+      port_sets {
+        key = "HTTP_PORTS"
+
+        port_set {
+          definition = ["80", "443"]
+        }
+      }
+    }
+
+    rules_source {
+      rules_string = join("\n", concat([for rule in local.rules_string : rule.tls], [for rule in local.rules_string : rule.http]))
+    }
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_networkfirewall_rule_group" "ip_rules" {
+  count = length(local.sorted_ip_rules) > 0 ? 1 : 0
+
+  capacity = var.ip_rules_capacity
+  name     = "${var.name}-ip-rules"
+  type     = "STATEFUL"
```

Comment:
tags and Add encryption_configuration reuse variables from aws_networkfirewall_firewall